extension/jdtls.ext/com.microsoft.gradle.bs.importer/src/com/microsoft/gradle/bs/importer/GradleBuildServerBuildSupport.java

```diff
@@ -310,6 +329,63 @@ private String getEclipseCompatibleVersion(String javaVersion) {
         return javaVersion;
     }
 
+    private void addDependencyJars(Map<String, IClasspathEntry> dependencyMap,
+            DependencyModulesResult dependencyModuleResult, boolean isTest) {
+        for (DependencyModulesItem item : dependencyModuleResult.getItems()) {
+            for (DependencyModule module : item.getModules()) {
+                if (!"maven".equals(module.getDataKind())) {
+                    continue;
+                }
+                List<Map> artifacts = (List<Map>) getValue((Map) module.getData(), "artifacts");
+                if (artifacts == null) {
+                    continue;
+                }
+
+                File artifact = null;
+                File sourceArtifact = null;
+                for (Map artifactData : artifacts) {
+                    String uri = (String) getValue(artifactData, "uri");
+                    if (uri == null) {
+                        continue;
+                    }
+                    String classifier = (String) getValue(artifactData, "classifier");
+                    try {
+                        File jarFile = new File(new URI(uri));
+                        if (classifier == null) {
+                            artifact = jarFile;
+                        } else if ("sources".equals(classifier)) {
+                            sourceArtifact = jarFile;
+                        }
+                    } catch (URISyntaxException e) {
+                        JavaLanguageServerPlugin.logException(e);
+                        continue;
+                    }
+                }
+
+                if (artifact == null) {
+                    continue;
+                }
+
+                List<IClasspathAttribute> attributes = new LinkedList<>();
+                if (isTest) {
+                    attributes.add(testAttribute);
+                }
+                if (!artifact.exists()) {
+                    attributes.add(optionalAttribute);
+                }
+
+                dependencyMap.putIfAbsent(artifact.getAbsolutePath(), JavaCore.newLibraryEntry(
```

Comment:
It seems no way to specify the javadoc, but it might be OK since the source usually contains javadoc as well.